### PR TITLE
Add metric of log

### PR DESCRIPTION
**Pull request by codings-dan**
### What changes are proposed in this pull request?

Add metric of log.

### CODE REVIEW

**Review by maobaolong — CHANGES_REQUESTED** [review]
Looks good overall, Please take a look at the failed CI checks, it seems like some style issues.

File: conf/log4j.properties [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 # May get overridden by System Property
 
-log4j.rootLogger=INFO, ${alluxio.logger.type}, ${alluxio.remote.logger.type}
+log4j.rootLogger=INFO, Console, ${alluxio.logger.type}, ${alluxio.remote.logger.type},EventCounter
```
Comment (LuQQiu) [inline conf/log4j.properties:14]: What user-facing changes will be introduced after changing this line?
I am a little worried about changing the rootLogger

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    public static long getInfo() {
+        return sCount.get(3);
+    }
+
+    /**
+     * add the number of corresponding level log.
+     * @param event event of generating log
+     * @return
+     */
+    public void append(LoggingEvent event) {
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:72]: Who calls append? is the append directly called by log4j internally every time when a new log is logged?

**Review by LuQQiu — COMMENTED** [review]
Thanks for adding the metric for logs! Those kinds of metrics are really helpful, just have some questions to understand the implementation

**Comment by LuQQiu** [comment]
Also please help fix the checkstyle issues which can be verified by running `mvn checkstyle:checkstyle` locally

File: conf/log4j.properties (hunk 1 above)
Comment (codings-dan) [reply conf/log4j.properties:14]: There is no change on the client side. This line of change can make the log output to the eventcounter class

File: core/common/src/main/java/alluxio/metrics/EventCounter.java (hunk 2 above)
Comment (codings-dan) [reply core/common/src/main/java/alluxio/metrics/EventCounter.java:72]: Yes, append is directly called by log4j every time a new log is generated

**Comment by maobaolong** [comment]
@codings-dan Do not use close&open to trigger the ci-checks restart, you can take a look at the failed test, if you think it isn't related to this PR, just left a comment to clarify and left the failed ci test for maintainer.

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+public class EventCounter extends AppenderSkeleton {
+  private static final int FATAL = 0;
+  private static final int ERROR = 1;
+  private static final int WARN = 2;
+  private static final int INFO = 3;
+  private static EventCounter.EventCounts sCount = new EventCounter.EventCounts();
+
+    /**
+     * the constructor of EventCounter.
+     */
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:31]: Indentation for all the comments is 2 spaces, same as the java code, please fix here as well as all the other places.
The description starts from a capital letter

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  public EventCounter() {
+  }
+
+    /**
+     * get the number of fatal log.
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:36]: Same indentation and capital letter issue here. 
Add a line between the description and @return 
you can see other methods inside the code base for example

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+     * @return the number of fatal log
+     */
+  public static long getFatal() {
+    return sCount.get(0);
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:40]: I saw you use 
```
  private static final int FATAL = 0;
  private static final int ERROR = 1;
  private static final int WARN = 2;
  private static final int INFO = 3;
```
then can you use those constants at all places?
like 
```
return sCount.get(FATAL)
```

**Review by LuQQiu — COMMENTED** [review]
Overall LGTM, left some code style comments

**Comment by LuQQiu** [comment]
@yuzhu Do you want to take another look at this PR?

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+ */
+public class EventCounter extends AppenderSkeleton {
+  private static final int FATAL = 0;
+  private static final int ERROR = 1;
+  private static final int WARN = 2;
+  private static final int INFO = 3;
+  private static EventCounter.EventCounts sCount = new EventCounter.EventCounts();
+
+  /**
+   * the constructor of EventCounter.
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:30]:
```suggestion
   * The constructor of EventCounter.
```

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+   * get the number of fatal log.
+   * @return the number of fatal log
+   */
+  public static long getFatal() {
+    return sCount.get(FATAL);
+  }
+
+  /**
+   * get the number of error log.
+   * @return the number of error log
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:45]:
```suggestion
   *
   * @return the number of error log
```

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+  /**
+   * get the number of fatal log.
+   * @return the number of fatal log
+   */
+  public static long getFatal() {
+    return sCount.get(FATAL);
+  }
+
+  /**
+   * get the number of error log.
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:44]:
```suggestion
   * Get the number of error log.
```

**Review by LuQQiu — COMMENTED** [review]
Please do the same for all other method descriptions

**Review by LuQQiu — APPROVED** [review]
LGTM! Thanks for adding the metrics of log. @yuzhu @apc999 do you guys want to take a final round of review?
If nobody reviews it after two days, this PR will be merged directly.

**Comment by maobaolong** [comment]
@LuQQiu These metrics are useful and works well in our cluster. Thanks @codings-dan

File: conf/log4j.properties (hunk 1 above)
Comment (maobaolong) [reply conf/log4j.properties:14]: I suggest to remove Console logger here, if who want to add this logger, just define `alluxio.logger.type`.

For `EventCounter `, an extra space between comma and letter E is necessary to keep style consistent.

Comment (maobaolong) [inline conf/log4j.properties:14]:
```suggestion
log4j.rootLogger=INFO, ${alluxio.logger.type}, ${alluxio.remote.logger.type}, EventCounter
```

File: conf/log4j.properties [hunk 9]
```diff
@@ -34,6 +34,9 @@ log4j.appender.Console.layout.ConversionPattern=%d{ISO8601} %-5p %c{1} - %m%n
 log4j.logger.org.apache.parquet.hadoop.InternalParquetRecordWriter=WARN
 log4j.logger.org.apache.parquet.hadoop.InternalParquetRecordReader=WARN
 
+#Appender for EventCounter
```
Comment (maobaolong) [inline conf/log4j.properties:37]:
```suggestion
# Appender for EventCounter
```

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+ */
+
+package alluxio.metrics;
+
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.Level;
+import org.apache.log4j.spi.LoggingEvent;
+
+/**
+ * A log4J Appender that simply counts logging events in three levels:
```
Comment (maobaolong) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:19]:
```suggestion
 * A log4J Appender that simply counts logging events in four levels:
```

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+ * fatal, error and warn. The class name is used in log4j.properties
```
Comment (maobaolong) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:20]:
```suggestion
 * FATAL, ERROR, WARN and INFO. The class name is used in log4j.properties
```

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+   */
+  public static long getInfo() {
+    return sCount.get(INFO);
+  }
+
+  /**
+   * Add the number of corresponding level log.
+   *
+   * @param event event of generating log
+   * @return
```
Comment (maobaolong) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:75]:
```suggestion
```

**Review by maobaolong — APPROVED** [review]
I approved this PR after you address my comments.

File: core/common/src/main/java/alluxio/metrics/EventCounter.java [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+  private static EventCounter.EventCounts sCount = new EventCounter.EventCounts();
+
+  /**
+   * The constructor of EventCounter.
+   */
+  public EventCounter() {
+  }
+
+  /**
+   * Get the number of fatal log.
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/EventCounter.java:36]: According to javaDoc convention, we use 3rd person (descriptive) 
can you help change the `Get` to `Gets` in the method description, similar for all other methods

**Review by LuQQiu — APPROVED** [review]
left one minor comment, thanks!

File: core/common/src/main/java/alluxio/metrics/EventCounter.java (hunk 13 above)
Comment (codings-dan) [reply core/common/src/main/java/alluxio/metrics/EventCounter.java:36]: ok,I will change it.

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**